pysr/regressor_sequence.py

```diff
@@ -0,0 +1,381 @@
+        raise ValueError("The length of `weights` must have shape (n_times,).")
+    if isinstance(variable_names, list) and len(variable_names) != X.shape[1]:
+        raise ValueError(
+            "The length of `variable_names` must be equal to the number of features in `X`."
+        )
+    if isinstance(X_units, list) and len(X_units) != X.shape[1]:
+        raise ValueError(
+            "The length of `X_units` must be equal to the number of features in `X`."
+        )
+    if y is not None:
+        raise ValueError("Recursive symbolic regression does not use `y`")
+    if y_units is not None:
+        raise ValueError("Recursive symbolic regression does not use `y_units`")
+    if Xresampled is not None:
+        raise ValueError("Recursive symbolic regression does not use `Xresampled`")
+
+
+class PySRSequenceRegressor(BaseEstimator):
+    """
+    High performance symbolic regression for time series data.
+    Based off of the `PySRRegressor` class, but with a preprocessing step for recurrence relations.
+
+    Parameters
+    ----------
+    recursive_history_length : int
+        The number of previous time points to use as input features.
+        For example, if `recursive_history_length=2`, then the input features
+        will be `[X[0], X[1]]` and the output will be `X[2]`.
+        This continues on for all X: [X[n-1], X[n-2]] to predict X[n].
+        Must be greater than 0.
+    Other parameters and attributes are inherited from `PySRRegressor`.
+    """
+
+    def __init__(
+        self,
+        recursive_history_length: int = 0,
+        **kwargs,
+    ):
+        self._regressor = PySRRegressor(**kwargs)
+        self.recursive_history_length = recursive_history_length
+
+    def _construct_variable_names(
+        self, n_features: int,
+        variable_names: Optional[List[str]]
+    ):
+        if not isinstance(variable_names, list):
+            if n_features == 1:
+                return [f"xt_{i}" for i in range(self.recursive_history_length, 0, -1)]
+            else:
+                return [
+                    f"x{i}t_{j}"
+                    for j in range(self.recursive_history_length, 0, -1)
+                    for i in range(n_features)
+                ]
+        else:
+            return [
+                i + "t_" + str(j)
+                for j in range(self.recursive_history_length, 0, -1)
+                for i in variable_names
+            ]
+
+    def fit(
+        self,
+        X,
+        y=None,
+        Xresampled=None,
+        weights=None,
+        variable_names: Optional[ArrayLike[str]] = None,
+        complexity_of_variables: Optional[
+            Union[int, float, List[Union[int, float]]]
+        ] = None,
+        X_units: Optional[ArrayLike[str]] = None,
+        y_units=None,
+    ) -> "PySRSequenceRegressor":
+        """
+        Search for equations to fit the time series dataset and store them in `self.equations_`.
```

Comment:
Instead of "time series" could it be called a "sequence" or something like that? I think it is more general than just time series datasets. You could have mathematical recurrence relationships for example – there's no sense of "time" there